src/pages/en/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -2,67 +2,884 @@
+
+### Migrating Tests
+
+As Astro outputs raw HTML, it is possible to write end-to-end tests using the output of the build step. Any end-to-end tests written previously might work out-of-the-box, if you have been able to match the markup of the older Gatsby site. Testing libraries such as Jest and React Testing Library can be imported and used in Astro to test your React components.
+
+See Astro's [testing guide](/en/guides/testing/) for more.
+
+## Reference: Convert to Astro Syntax
+
+The following are some examples of Gatsby-specific syntax that you will need to convert to Astro. See more [differences between Astro and JSX](/en/core-concepts/astro-components/#differences-between-astro-and-jsx) in the guide to writing Astro components.
+
+### Gatsby Links to Astro
+
+Convert any Gatsby `<Link to="">`, `<NavLink>` etc. components to HTML `<a href="">` tags. 
+
+Astro does not use any special component for links, although you are welcome to build custom link components. You can then import and use this `<Link>` just as you would any other component.
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Gatsby Imports to Astro
+
+If necessary, update any file imports to reference relative file paths exactly. This can be done using [import aliases](/en/guides/typescript/#import-aliases), or by writing out a relative path in full (e.g. `../../layouts/Layout.astro`). 
+
+Note that `.astro` and several other file types must be imported with their full file extension.
+
+### Gatsby Children Props to Astro
+
+Convert any instances of `{children}` to an Astro `<slot />`. Astro does not need to receive `{children}` as a function prop and will automatically render child content in a `<slot />`. 
+
+To access specific attributes passed to your component (e.g. `<Layout title="About Me"/>`), use `Astro.props`.
+
+### Gatsby Styling to Astro
+
+Convert any inline style objects (`style = {{fontWeight: "bold", }}`) to inline HTML style attributes (`style="font-weight:bold;"`). Or, use an Astro `<style>` tag for scoped CSS styles.
 
-Bring your existing Markdown (or MDX, with our optional integration) files as content to [create Markdown or MDX pages](/en/guides/markdown-content/).
+Global styling is achieved in Gatsby using CSS imports in `gatsby-browser.js`. In Astro, you will import `.css` files directly into a main layout component to achieve global styles.
 
-While file-based routing and layout components are similar in Astro, you may wish to read about [Astro's project structure](/en/core-concepts/project-structure/) to learn where files should be located. For example, you will need to rename your Gatsby static assets folder from `static/` to `public/`.
+### Gatsby Image Plugin to Astro
```

Comment:
I think this needs to be expanded. It is one of the most convenient and important parts of Gatsby. 

See my comments in above/below and in the other issue.

But also, there should IMO be a list of gatsby image features and their corresponding Astro Image features. Like the lazy loading feature and resize options.